python/lsst/obs/rubinGenericCamera/translator.py

```diff
@@ -0,0 +1,256 @@
+            return self.quantity_from_card("EXPTIME", u.s)
+
+        # A missing or undefined EXPTIME is problematic. Set to -1
+        # to indicate that none was found.
+        self.log.warning("%s: Insufficient information to derive exposure time. Setting to -1.0s",
+                         self._log_prefix)
+        return -1.0 * u.s
+
+    @cache_translation
+    def to_dark_time(self):             # N.b. defining this suppresses a warning re setting from exptime
+        if "DARKTIME" in self._header:
+            darkTime = self._header["DARKTIME"]
+            self._used_these_cards("DARKTIME")
+            return (darkTime, dict(unit=u.s))
+        return self.to_exposure_time()
+
+
+class StarTrackerTranslator(RubinGenericCameraTranslator):
+    name = None                         # must be specialised
+    """Name of this translation class"""
+
+    supported_instrument = None         # must be specialised
+    """Supports the Rubin Star Tracker instrument."""
+
+    @classmethod
+    def _is_startracker(cls, header, filename=None):
+        """Indicate whether the supplied header comes from a starTracker
+
+        Parameters
+        ----------
+        header : `dict`-like
+            Header to convert to standardized form.
+        filename : `str`, optional
+            Name of file being translated.
+
+        Returns
+        -------
+        (isStarTracker, camId) : (`True`, `int`) or (`False`, `None`)
+            isStarTracker is `True` iff header comes from a starTracker
+            camId is the cameraID, e.g. 101 for the wide-field startracker,
+            102 for the narrow-field startracker
+        """
+        if "INSTRUME" not in header or header["INSTRUME"] != "StarTracker" or "OBSID" not in header:
+            return (False, None)
+
+        camId = int(header["OBSID"][2:5])
+
+        return (True, camId)
+
+
+class StarTrackerNarrwTranslator(StarTrackerTranslator):
+    name = "StarTrackerNarrw"
+    """Name of this translation class"""
+
+    supported_instrument = "StarTrackerNarrw"
+    """Supports the Rubin Star Tracker narrow-field instrument."""
+
+    @classmethod
+    def can_translate(cls, header, filename=None):
+        """Indicate whether this translation class can translate the
+        supplied header.
+
+        Parameters
+        ----------
+        header : `dict`-like
+            Header to convert to standardized form.
+        filename : `str`, optional
+            Name of file being translated.
+
+        Returns
+        -------
+        can : `bool`
+            `True` if the header is recognized by this class. `False`
+            otherwise.
+        """
+
+        isStarTracker, camId = cls._is_startracker(header, filename=None)
+
+        return isStarTracker and camId == 102
+
+    @cache_translation
+    def to_instrument(self):
+        return "StarTrackerNarrw"
+
+    @cache_translation
+    def to_detector_serial(self):
+        return "00:0f:31:03:ae:60"
```

Comment:
I think it's meant to be allowed for const_map to work in subclasses so you can use:

```python
_const_map = {
    "detectors_serial": "00:0f:31:03:ae:60",
    "instrument": "StarTrackerNarrw",
}
```